app/controllers/tracks_controller.rb

```diff
@@ -26,9 +26,10 @@ def create
           waveform_url: track_data['waveform_url'],
           bpm: track_data['bpm']
           )
-        next unless track.save
-      end
-      render plain: 'AJAX OK'
+          next unless track.save
+          tracks << track
```

Comment:
Oh i think i may know what's bothering me, are you using `#each` to accumulate to an array?  That's what `#map` is for.